### PR TITLE
Converting `anyhow` to `this-error` for `re_renderer render_pipeline_pool`

### PR DESCRIPTION
**Pull request by chinepun**
### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3812) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/3812)
- [Docs preview](https://rerun.io/preview/481ea711d07faa13e1c45e90eac755ba44d47caa/docs) 
- [Examples preview](https://rerun.io/preview/481ea711d07faa13e1c45e90eac755ba44d47caa/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs [hunk 1]
```diff
@@ -102,23 +101,40 @@ pub struct RenderPipelineDesc {
...
+        let pipeline_layout =
+            if let Ok(pipeline_layout) = pipeline_layouts.get_resource(self.pipeline_layout) {
+                Some(&pipeline_layout.layout)
+            } else {
+                return Err(RenderPipelineError::PipelineLayoutNotPresent());
+            };
+
+        let Ok(vertex_shader_module) = shader_modules.get(self.vertex_handle) else {
+            re_log::error!("referenced vertex shader not found");
+            return Err(RenderPipelineError::VertexShaderNotFound());
```
Comment (emilk) [inline crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs:132]: Don't both log AND return the error - that will likely get it logged twice!

See https://github.com/rerun-io/rerun/blob/main/CODE_STYLE.md#error-handling-and-logging for more

File: crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs [hunk 2]
```diff
@@ -210,13 +226,7 @@ impl GpuRenderPipelinePool {
...
                     Some(sm)
                 }
-                Err(err) => {
-                    re_log::error!(
-                        err = re_error::format(err),
-                        "couldn't recompile render pipeline"
-                    );
-                    None
-                }
+                Err(_) => None,
```
Comment (emilk) [inline crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs:229]: Don't ignore errors - either return them or log them

```suggestion
                Err(err) => {
                    re_log::error!("Failed to compile render pipeline: {}", re_error::format(err));
                    None
                }
```

File: crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs [hunk 3]
```diff
@@ -102,23 +101,40 @@ pub struct RenderPipelineDesc {
...
-
-        let vertex_shader_module = shader_modules
-            .get(self.vertex_handle)
-            .context("referenced vertex shader not found")?;
-        let fragment_shader_module = shader_modules
-            .get(self.fragment_handle)
-            .context("referenced fragment shader not found")?;
+    ) -> Result<wgpu::RenderPipeline, RenderPipelineError> {
+        let pipeline_layout =
+            if let Ok(pipeline_layout) = pipeline_layouts.get_resource(self.pipeline_layout) {
```
Comment (emilk) [inline crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs:124]: If you use `let Ok(…)` you miss out the error in the `else` branch. Instead, return the inner error using `.map_err(RenderPipelineError::PipelineLayoutNotPresent)?`

here and everywhere

File: crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs [hunk 4]
```diff
@@ -210,13 +226,7 @@ impl GpuRenderPipelinePool {
                     re_log::info!(label = desc.label.get(), "recompiled render pipeline");
                     Some(sm)
                 }
-                Err(err) => {
-                    re_log::error!(
-                        err = re_error::format(err),
-                        "couldn't recompile render pipeline"
-                    );
-                    None
-                }
```
Comment (Wumpf) [inline crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs:219]: Don't remove this! This literally does the logging you unnecessarily added in `create_render_pipeline` ;)

**Review by emilk — DISMISSED** [review]
Thanks for the PR!

However: Your code discards a lot of the error messages, which is a big no-no! See `CODE_STYLE.md` for details

You also comited two extra (huge) files - please remove 😅

**Comment by Wumpf** [comment]
ah Emil beat me to it once more. My fault, I didn't put myself as reviewer before starting to review.

File: crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs (hunk 2 above)
Comment (chinepun) [reply crates/re_renderer/src/wgpu_resources/render_pipeline_pool.rs]: I can't seem to find a convenient way to  log the `err` here

**Review by Wumpf — APPROVED** [review]

**Pull request merged**